### ISSUE TITLE
Question on `nextElement` implementation

### ISSUE THREAD

**Issue by hipstermojo** [mozilla/readability#629]
This function is used to get the next possible element and may return null however the code:
```js
this.REGEXPS.whitespace.test(next.textContent)
```
allows for the possibility of text nodes also being returned. Is this the intended purpose of the function; to return both element and text nodes?

**Comment by gijsk** [comment]
Yes, it looks like it can return both. The code at https://github.com/mozilla/readability/blame/d5eea06a0095b3138dbd1f6233f656d690200509/Readability.js#L580 makes this look intentional: if there's markup like:

```html
<p>
Some text
<br><br>
<br>
Some more text
</p>
```

it intends to loop over the `<br>` tags, and the whitespace inbetween, but then stop short of the "Some more text" text node (as it won't match the whitespace regex).

It'd probably be a good idea to rename this so it doesn't imply it only returns elements.

**Comment by ankushduacodes** [comment]
Hi @gijsk, @hipstermojo, I would love to work on this issue, but I am very new to open source and this will only be my 3rd or 4th contribution, Is this a beginner-friendly issue, if yes that I would love to take on this issue and if not can you please recommend a good beginner-friendly issue that I can work on.
Thank you.

**Comment by gijsk** [comment]
Hi @ankushduacodes , sure, you could pick this up. Fixing it would involve renaming the `nextElement` method in Readability.js to `nextNode`.

**Comment by ankushduacodes** [comment]
@gijsk Could you please give a starting point from which file should start or does the order not matter?
and I will be just renaming nextElement method to nextNode right or is there something more that I should be aware of?

**Comment by gijsk** [comment]
I think the method is only used from within the `Readability.js` file. And I don't think there's anything else to be aware of - once you're done, it'd be good to make sure tests still pass, you can check with `npm test`.

**Comment by ankushduacodes** [comment]
For sure @gijsk,
Thank you for the tip

**Comment by gijsk** [comment]
This was fixed in #637 . \o/
